External/FEXCore/Source/Utils/Allocator.cpp

```diff
@@ -240,6 +248,22 @@ namespace FEXCore::Allocator {
           STEAL_LOG("[%d] ParseEnd; RegionBegin: %016lX RegionEnd: %016lX\n", __LINE__, RegionBegin, RegionEnd);
 
           State = ScanEnd;
+
+          // If the previous map's ending and the region we just parsed overlaps the stack then we need to save the stackmapping.
+          // Otherwise we will only have ~144KB stack size which crashes quickly.
+          if (PreviousMapEnd <= StackLocation_u64 && RegionEnd > StackLocation_u64) {
```

Comment:
How did you come up with this check? Specifically, why does `PreviousMapEnd <= StackLocation_u64` check that the previous map overlaps with the stack? (My understanding being that the stack ends at `StackLocation_u64` and begins at an address higher than that)

Reply:
This is absolutely bonkers how the logic in this function works.
First step is that it parses the region below the stack space, mmap'ing it with no privileges. Saving the `PreviousMapEnd` variable to the start of the mapping address.

It then parses the stack mapping in the `State == ParseEnd` step.
This is where the `if (PreviousMapEnd <= StackLocation_u64 && RegionEnd > StackLocation_u64)` comparison comes in.
First check ensuring that the previous mapping end address is less than the current stack pointer. Ensuring we haven't progressed beyond the stack comparison point.

Second check ensuring that the /current/ parsed region is higher than the current stack pointer, ensuring that this full region is the stack region.
As long as everything is allocated in order like expected then this holds true that it is the stack.

As an additional step of safety, when we pull the last allocation off the vector of allocations, we double check that `Ptr+Size` is the start of the mapping, ensuring the regions are contiguous with each other.

Absolute nightmare, but we'll rewrite this code to be easier to parse.